bake/hclparser/stdlib.go

```diff
@@ -254,6 +256,115 @@ func timestampFunc() function.Function {
 	})
 }
 
+// rfc3339ParseFunc, given an RFC3339 timestamp string, will parse and return an
+// object representation of that date and time.
+//
+// This function is similar to the `rfc3339_parse` function in Terraform:
+// https://registry.terraform.io/providers/hashicorp/time/latest/docs/functions/rfc3339_parse
+func rfc3339ParseFunc() function.Function {
+	return function.New(&function.Spec{
+		Description: `Given an RFC3339 timestamp string, will parse and return an object representation of that date and time.`,
+		Params: []function.Parameter{
+			{
+				Name:        "timestamp",
+				Description: "RFC3339 timestamp string to parse",
+				Type:        cty.String,
+			},
+		},
+		Type: function.StaticReturnType(cty.Object(map[string]cty.Type{
+			"year":         cty.Number,
+			"year_day":     cty.Number,
+			"day":          cty.Number,
+			"month":        cty.Number,
+			"month_name":   cty.String,
+			"weekday":      cty.Number,
+			"weekday_name": cty.String,
+			"hour":         cty.Number,
+			"minute":       cty.Number,
+			"second":       cty.Number,
+			"unix":         cty.Number,
+			"iso_year":     cty.Number,
+			"iso_week":     cty.Number,
+		})),
+		Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
+			ts := args[0].AsString()
```

Comment:
Is it possible for `args` to be empty?

Reply:
No it would fail earlier because an arg is required for this func